core/src/main/java/org/elasticsearch/rest/BytesRestResponse.java

```diff
@@ -135,9 +139,27 @@ private static XContentBuilder build(RestChannel channel, RestStatus status, Exc
         }
 
         XContentBuilder builder = channel.newErrorBuilder().startObject();
+        builder.field(STATUS, status.getStatus());
         ElasticsearchException.generateFailureXContent(builder, params, e, channel.detailedErrorsEnabled());
-        builder.field("status", status.getStatus());
         builder.endObject();
         return builder;
     }
+
+    public static ElasticsearchException errorFromXContent(XContentParser parser) throws IOException {
+        XContentParser.Token token = parser.nextToken();
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);
+
+        ElasticsearchException exception = null;
+
+        String currentFieldName = parser.currentName();
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+            }
+            if (STATUS.equals(currentFieldName) == false) {
+                exception = ElasticsearchException.failureFromXContent(parser);
+            }
```

Comment:
given that the high level client will simply throw the exception, the status code gets lost and that is not good. I would  be ok also with having a way to provide the status code when creating the exception rather than always detect it. Then the fromXContent could accept it as an argument. Or even better we could parse it rather than ignore it. Thoughts?